docker/Dockerfile

```diff
@@ -1,42 +1,26 @@
-FROM ubuntu:trusty
-
-RUN apt-get update && apt-get install -y git
-RUN apt-get install -y software-properties-common
-RUN add-apt-repository ppa:joyard-nicolas/ffmpeg -y
-RUN apt-get -qq update
-RUN apt-get install -y ffmpeg tesseract-ocr
-RUN apt-get install -y wget
-RUN apt-get install -y graphviz libgraphviz-dev pkg-config
-
-ENV MINICONDA_URL "https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh"
-ENV CONDA_DEPS "pip jupyter pytest numpy pandas pillow==2.9.0 requests scipy nltk six tqdm seaborn matplotlib pathos tensorflow contextlib2"
-WORKDIR /root
-ENV MINICONDA $HOME/miniconda
-ENV PATH "$MINICONDA/bin:$PATH"
-RUN wget $MINICONDA_URL -O miniconda.sh;
-RUN bash miniconda.sh -b -f -p $MINICONDA;
-RUN hash -r
-RUN conda info -a
-RUN conda config --set always_yes yes --set changeps1 no
-RUN conda config --add channels conda-forge
-RUN conda update -y conda
-RUN conda create -y -n py35 python=3.5 $CONDA_DEPS
-
-RUN /bin/bash -c "source /miniconda/envs/py35/bin/activate py35 \
-  && pip install --upgrade --ignore-installed setuptools \
-  && pip install python-magic coveralls pytest-cov pygraphviz pysrt xlrd clarifai pytesseract moviepy==0.2.2.13 SpeechRecognition IndicoIo sklearn python-twitter gensim oauth2client google-api-python-client google-compute-engine librosa ipython"
-
-RUN /bin/bash -c 'source /miniconda/envs/py35/bin/activate py35 \
- && python -c "import imageio; imageio.plugins.ffmpeg.download()"'
-
-RUN git clone https://github.com/tyarkoni/pliers.git
-WORKDIR /root/pliers
-RUN /bin/bash -c 'source /miniconda/envs/py35/bin/activate py35 \
- && python setup.py install'
-RUN echo "if (tty -s); then \n\
-    source /miniconda/envs/py35/bin/activate py35\n\
-fi" >> /root/.bashrc
-RUN /bin/bash -c 'source /miniconda/envs/py35/bin/activate py35 \
- && python -m pliers.support.download'
-RUN apt-get install -y curl
-CMD ["/bin/bash"]
+FROM python:3.6-slim
+ARG DEBIAN_FRONTEND="noninteractive"
+WORKDIR /opt/pliers
+COPY . .
+RUN apt-get update -qq \
+    && tmp_pkgs="cmake gcc g++ libc6-dev libgraphviz-dev libmagic-dev make" \
+    && apt-get install -yq --no-install-recommends \
+        ffmpeg \
+        graphviz \
+        libmagic1 \
+        tesseract-ocr \
+        $tmp_pkgs \
+    && pip install --no-cache-dir \
+        --requirement requirements.txt \
+        --requirement optional-dependencies.txt \
+        ipython \
+        notebook \
+    && python -m spacy download en_core_web_sm \
```

Comment:
This may also be needed:
`python -m pliers.support.download`

Also, I think it might be better to split this huge `RUN` command into several smaller `RUN` commands, rather than one big statement. Docker build caching might work better (unless you have a reason for doing it this way?

Reply:
I was debating whether to add `python -m pliers.support.download`. That downloads things into the current `$HOME`, so if someone used the container as a different user (e.g., in singularity), those assets wouldn't be available. But I can add that download, and users can re-download if they do end up using the container as a different user.

I suppose parts of the `RUN` instruction can be split, but I wrote it this way because I download and later delete temporary dependencies (e.g.,`gcc`). If the download and deletion were in separate `RUN` instructions, the size of the resulting image would be larger.

I also ran `pytest` in the container, and the tests that don't require payment passed.

Reply:
Cool, that's a good reason to keep it as is then!

Just add the support download (I think its best if the image works for most users out of the box)

Reply:
Sounds good. And I think I did need to run `python -m pliers.support.download` for the tests to pass.